lib/Sema/TypeCheckType.cpp

```diff
@@ -2371,7 +2371,7 @@ bool swift::diagnoseMissingOwnership(ASTContext &ctx, DeclContext *dc,
   if (options.hasBase(TypeResolverContext::EnumElementDecl))
     return false; // no need for ownership in enum cases.
 
-  if (!ty->isNoncopyable(dc))
+  if (!isInterfaceTypeNoncopyable(ty, dc->getGenericEnvironmentOfContext()))
```

Comment:
use options.getGenericSignature().getGenericEnvironment() instead -- that will fix SIL mode.